Extension/src/Debugger/configurationProvider.ts

```diff
@@ -467,22 +476,23 @@ export class DebugConfigurationProvider implements vscode.DebugConfigurationProv
                         }
                         const compilerDirname: string = path.dirname(compilerPath);
                         const debuggerPath: string = path.join(compilerDirname, debuggerName);
-                        if (isWindows) {
-                            newConfig.miDebuggerPath = debuggerPath;
-                            return resolve(newConfig);
-                        } else {
-                            fs.stat(debuggerPath, (err, stats: fs.Stats) => {
-                                if (!err && stats && stats.isFile()) {
-                                    newConfig.miDebuggerPath = debuggerPath;
-                                } else {
-                                    newConfig.miDebuggerPath = path.join("/usr", "bin", debuggerName);
+                        if (!await util.checkFileExists(debuggerPath)) {
+                            if (!isWindows) {
+                                const newDebuggerPath: string = path.join("/usr", "bin", debuggerName);
+                                if (await util.checkFileExists(debuggerPath)) {
```

Comment:
It seems like this should be `if (await util.checkFileExists(newDebuggerPath)) {`